src/frontend/src/pages/FinancePage/FinanceComponents/PendingAdvisorListModal.tsx

```diff
@@ -0,0 +1,110 @@
+import { Controller, useForm } from 'react-hook-form';
+import { useToast } from '../../../hooks/toasts.hooks';
+import { useSendPendingAdvisorList } from '../../../hooks/finance.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import NERFormModal from '../../../components/NERFormModal';
+import { Box } from '@mui/system';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { TextField } from '@mui/material';
+import { ChangeEvent, useState } from 'react';
+
+const schema = yup.object().shape({
+  saboNumbers: yup.array().of(yup.number().required('SABO Number is required')).required('SABO Numbers are required')
+});
+
+interface PendingAdvisorModalProps {
+  open: boolean;
+  saboNumbers: number[];
+  onHide: () => void;
+}
+const PendingAdvisorModal: React.FC<PendingAdvisorModalProps> = ({ open, saboNumbers, onHide }) => {
+  const { mutateAsync: sendPendingAdvisorList, isLoading: sendPendingAdvisorListIsLoading } = useSendPendingAdvisorList();
+
+  const {
+    handleSubmit,
+    reset,
+    control,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      saboNumbers: saboNumbers
+    }
+  });
+
+  const toast = useToast();
+
+  if (sendPendingAdvisorListIsLoading) return <LoadingIndicator />;
+
+  const onSubmit = async (data: { saboNumbers: number[] }) => {
+    try {
+      await sendPendingAdvisorList(data.saboNumbers);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        toast.error(error.message);
+      }
+    }
+  };
+
+  // Custom input component for the TextField
+  const CommaSeparatedNumbersInput = ({
```

Comment:
you can move this component to the top of the file, and add errors as a prop. Makes it cleaner/ more reusable in other places if need be later